scvi/metrics/imputation.py

```diff
@@ -1,18 +1,118 @@
+import copy
+
+import matplotlib.pyplot as plt
 import numpy as np
 import torch
+from scipy.stats import kde
+
+from scvi.dataset import GeneExpressionDataset
+from scvi.dataset.data_loaders import DataLoaderWrapper
+
+
+def imputation(infer, name, rate=0.1, n_samples=1, n_epochs=1, corruption="uniform"):
+    corrupted_data = copy.deepcopy(infer.gene_dataset.X)
```

Comment:
Not sure if that is memory efficient. (Before we did drop "on the fly", but there was no training of the model on the corrupted data, just forward pass on corrupted batches)

Reply:
Won't that double the memory use of the program? Does it work to "corrupt" the training data on the fly, in the training loop?

Reply:
Yes I agree, it is suboptimal in terms of space but not in term of clarity I think.

I don't see a very clean way to do it on the fly. We have to generate anyway 10% of new data. We could maybe somehow get the indices of the shuffled dataloaders in the training loop, but it would be computationally inefficient to overwrite the input for the corresponding indices at each loop iteration.

For a large dataset like brain large, romain only took a subsample of the dataset to run the imputation benchmark.